src/blocks/testimonials/testimonial/edit.js

```diff
@@ -0,0 +1,346 @@
+/**
+ * External dependencies.
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import { isBlobURL } from '@wordpress/blob';
+import { mediaUpload } from '@wordpress/editor';
+import { usePrevious } from '@wordpress/compose';
+import { BlockIcon, getColorClassName, MediaUpload, MediaUploadCheck, RichText } from '@wordpress/block-editor';
+import { Button, ButtonGroup, DropZone, Spinner } from '@wordpress/components';
+import { closeSmall, image } from '@wordpress/icons';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import Controls from '../controls';
+import Inspector from './inspector';
+
+/**
+ * Set and export block values.
+ */
+const MIN_IMAGE_SIZE = 50;
+const MAX_IMAGE_SIZE = 250;
+
+export { MIN_IMAGE_SIZE, MAX_IMAGE_SIZE };
+
+const Edit = ( props ) => {
+	const {
+		attributes,
+		className,
+		isSelected,
+		setAttributes,
+	} = props;
+
+	const {
+		customBackgroundColor,
+		customBubbleBackgroundColor,
+		customBubbleTextColor,
+		customTextColor,
+		backgroundColor,
+		bubbleBackgroundColor,
+		bubbleTextColor,
+		headingLevel,
+		name,
+		role,
+		showImage,
+		showRole,
+		styleName,
+		text,
+		textColor,
+	} = attributes;
+
+	const [ url, setUrl ] = useState( attributes.url || '' );
+
+	const prevShowRole = usePrevious( showRole );
```

Comment:
@jlemay-godaddy why do we need so many conditions here with usePrevious. The control themselves are not able to modify the attributes directly?